src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -988,7 +991,7 @@ public void InvalidateCacheEntry(ActivationAddress activationAddress, bool inval
         /// <returns></returns>
         public SiloAddress GetPrimaryForGrain(GrainId grain)
```

Comment:
`GetPrimaryForGrain` can now also return `null`, which needs to be accounted for by its callers.